Gems/ROS2/Code/Source/RobotImporter/Bus/RobotImporterBus.h

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+#pragma once
+
+#include "AzCore/EBus/Policies.h"
+#include "AzCore/std/string/string.h"
+#include <AzCore/Component/EntityId.h>
+#include <AzCore/EBus/EBus.h>
+#include <AzCore/RTTI/BehaviorContext.h>
+#include <AzCore/Serialization/SerializeContext.h>
+
+namespace ROS2
+{
+    class RobotImporterRequest : public AZ::EBusTraits
+    {
+    public:
+
+        static constexpr AZ::EBusHandlerPolicy HandlerPolicy = AZ::EBusHandlerPolicy::Single;
+        static constexpr AZ::EBusAddressPolicy AddressPolicy = AZ::EBusAddressPolicy::Single;
+
+        //! Generate prefab from urdf file, it is asynchronous call.
+        //! @param filePath The path of the urdf file
+        //! @param importAssetWithUrdf If true, the assets referenced in the urdf file will be imported
+        //! @param useArticulation If true, the prefab will be generated with articulation
+        virtual void GeneratePrefabFromFile(const AZStd::string_view filePath, bool importAssetWithUrdf, bool useArticulation) = 0;
+
+        //! Check if the prefab generation is done.
+        virtual bool IsDone() = 0;
+
+        //! Check if the prefab generation is successful.
+        virtual bool IsSuccessful() = 0;
+    };
+
+    using RobotImporterRequestBus = AZ::EBus<RobotImporterRequest>;
+
+    class RobotImporterRequestHandler
```

Comment:
I am not sure if RobotImporterRequestHandler is needed, please consider removing it and add reflection to RobotImporterRequestBus